doc/plugins.md

```diff
@@ -28,7 +28,7 @@ The list of plugins:
 
 *   [`rehype-accessible-emojis`](https://github.com/GaiAma/Coding4GaiAma/tree/HEAD/packages/rehype-accessible-emojis)
     — make emojis accessible adding role & aria-label
-*   [`rehype-add-classes`](https://github.com/martypdx/rehype-add-classes)
+*   [`rehype-class-names`](https://github.com/riderjensen/rehype-class-names)
```

Comment:
- The list is sorted alphabetically, can you fix that?
- This plugin has virtually zero downloads, can you maintain it for years to come/a while?
- `className` **must not** be a string, it’s an array of strings: https://github.com/syntax-tree/hast#propertyvalue
- Perhaps you can use https://github.com/syntax-tree/hast-util-classnames to add/remove classes in a smarter way?

Reply:
> * The list is sorted alphabetically, can you fix that?

Done.

> * This plugin has virtually zero downloads, can you maintain it for years to come/a while?

I don't maintain either package. But I use them. See https://github.com/nasa-gcn/gcn.nasa.gov/pull/1200.

> * `className` **must not** be a string, it’s an array of strings: https://github.com/syntax-tree/hast#propertyvalue

I don't understand this comment.

> * Perhaps you can use https://github.com/syntax-tree/hast-util-classnames to add/remove classes in a smarter way?

How would I do that in a context where I have the opportunity only to specify rehype plugins? https://github.com/nasa-gcn/gcn.nasa.gov/blob/main/remix.config.mjs#L13-L21

Reply:
> I don't maintain either package.

Ah, I was under the impression you made it.

> I don't understand this comment.

These lines are wrong: https://github.com/riderjensen/rehype-class-names/blob/768c1aaf0574c4b726ead09cede18d8cddbd766d/index.js#L9-L11.
`className` needs to be an array of strings. Not a string.

> How would I do that in a context where I have the opportunity only to specify rehype plugins?

I was reviewing the plugin you are adding to this list.

Reply:
Reviewing your nasa code, you are passing options in a rather weird way.

Use:

```js
      [rehypeExternalLinks, {rel: 'external', target: false}],
      [
        rehypeAddClasses,
        {
          a: 'usa-link',
          ol: 'usa-list',
          p: 'usa-paragraph',
          table: 'usa-table',
          ul: 'usa-list',
        }
      ],
      [rehypeAutolinkHeadings, {behavior: 'wrap'}],
```

Reply:
> Reviewing your nasa code, you are passing options in a rather weird way.

I'm not sure where I got the example that I based that on, but I think that they may have done that to expose the arguments of each plugin for autocompletion with TypeScript.

Reply:
> `className` needs to be an array of strings. Not a string.

rehype-add-classes seems to set classNames as a string, not an array of strings. Why does it appear to work?

Reply:
- You can add for example `/** @type {import('rehype-autolink-headings').Options} */` to an options object to check it. This is quite the same as the next point: it’s probably going to fail at some point or with some other plugins, please don’t do it.
- “Work” is complex. It doesn’t crash, yet. Some plugins accept it, currently. Other plugins fail. It’s documented that it should be arrays of strings, that plugin should do that, too.

Reply:
Where is that documented? I couldn't find that in the hast README.

Reply:
See:
a) example in `Element`: https://github.com/syntax-tree/hast#element
b) example in `PropertyValue`: https://github.com/syntax-tree/hast#propertyvalue